packages/astro/src/vite-plugin-build-css/index.ts

```diff
@@ -156,6 +176,34 @@ export function rollupPluginAstroBuildCSS(options: PluginOptions): VitePlugin {
 				.replace(/\./g, '\\.');
 			const emptyChunkRE = new RegExp(opts.format === 'es' ? `\\bimport\\s*"[^"]*(?:${emptyChunkFiles})";\n?` : `\\brequire\\(\\s*"[^"]*(?:${emptyChunkFiles})"\\);\n?`, 'g');
 
+			// Crawl the module graph to find CSS chunks to create
+			if(!legacy) {
+				for(const id of this.getModuleIds()) {
```

Comment:
if you move this into a separate function, it might be a bit easier to document / more self-documenting. Normally I'd call this a nit, but this sounds big & important enough that more documentation would be a good idea.

```js
if (!legacy) {
  this.addImportedStyles()
}

// ...

/** Use this comment to talk about why this is required **/
addImportedStyles() {

}
```